components/authz-service/server/v2/policy.go

```diff
@@ -663,26 +663,36 @@ func (s *policyServer) EngineUpdateInterceptor() grpc.UnaryServerInterceptor {
 			return nil, err
 		}
 
-		// ignore anything not for this service
-		if !strings.HasPrefix(info.FullMethod, "/chef.automate.domain.authz.v2.Policies/") {
+		// ignore anything not related to the OPA store.
+		if !strings.HasPrefix(info.FullMethod, "/chef.automate.domain.authz.v2.Policies/") &&
+			!strings.HasPrefix(info.FullMethod, "/chef.automate.domain.authz.v2.Projects/") {
 			return resp, nil
 		}
 
-		// Note: the set of methods we _do not want to update_ our engine store
-		// is the smaller one. Also, updating too often is not as bad as
-		// updating too little.
 		switch info.FullMethod {
-		case "/chef.automate.domain.authz.v2.Policies/ListPolicies",
-			"/chef.automate.domain.authz.v2.Policies/ListRoles",
-			"/chef.automate.domain.authz.v2.Policies/GetPolicy",
-			"/chef.automate.domain.authz.v2.Policies/GetRole",
-			"/chef.automate.domain.authz.v2.Policies/ListPolicyMembers",
-			"/chef.automate.domain.authz.v2.Policies/GetPolicyVersion":
-			// do nothing
-		default:
+		// Important! Any new endpoint that requires refreshing the OPA cache must be added here.
+		case "/chef.automate.domain.authz.v2.Policies/ReplacePolicyMembers",
+			"/chef.automate.domain.authz.v2.Policies/CreatePolicy",
+			"/chef.automate.domain.authz.v2.Policies/DeletePolicy",
+			"/chef.automate.domain.authz.v2.Policies/UpdatePolicy",
+			"/chef.automate.domain.authz.v2.Policies/MigrateToV2",
+			"/chef.automate.domain.authz.v2.Policies/ResetToV1",
+			"/chef.automate.domain.authz.v2.Policies/CreateRole",
+			"/chef.automate.domain.authz.v2.Policies/DeleteRole",
+			"/chef.automate.domain.authz.v2.Policies/UpdateRole",
+			"/chef.automate.domain.authz.v2.Policies/RemovePolicyMembers",
+			"/chef.automate.domain.authz.v2.Policies/AddPolicyMembers",
+			"/chef.automate.domain.authz.v2.Policies/PurgeSubjectFromPolicies":
 			if err := s.updateEngineStore(ctx); err != nil {
 				return nil, status.Errorf(codes.Internal, "error updating engine store: %s", err.Error())
 			}
+		default:
+			// do nothing
+		}
+
+		s.log.Debugf("Initiating store update for %s", info.FullMethod)
```

Comment:
❓ Why would we have done this? Above, we selectively run the engine update for specific methods, and after the "big switch", we _do it again regardless_? Am I missing something? 😕

Reply:
😮 No, don't think you are missing anything... Clearly needs fixing.

Reply:
👉 https://github.com/chef/automate/pull/467/commits/99840f085dc0ff630ab4cb08a59699e3f5c38fb1 wrapped into #467